Engineering/Entities in Domain Driven Design.md

```diff
@@ -0,0 +1,47 @@
+---
+tags: engineering, Domain-Driven Design
+author: M.Vu Cuong(Jim)
+---
+
+# Entities in Domain-Driven Design
+
+## Recap on DDD
+A software design focusing on the **Domain**, one of the keys to a program success, by removing *communication lag* between **Developers** and **Domain Experts**, *separate important Domain-specific components* for maintenance and scalability
+
+## A) Entities Defined by **Identity**
+
+*ID or Identity, identify 2 objects with the same attributes* (2 persons with the same name, DoB but different ID card number).
+
+## B) Entities Defined by **Thread of continuity**
+
+*A set of fingerprints or traces of an object when going through multiple systems.*
+
+## Example
+
+### Real case world view
+
+- **Banking(A)**, a banknote(paper money) will have an ID that identifies it on the digital systems or a physical log book even if all of its attributes are the same as another banknote(value, printing date, printing batch, etc..)
+
+- **Online shopping service(A, B)**, an order has ID. 
+    - After checkout, the system sends the order ID to the payment service.
```

Comment:
If we end the sentence with a period, should we keep it consistent for other sentences?